classes/Cart.php

```diff
@@ -3704,7 +3704,7 @@ public function getTotalWeight($products = null)
             $this->updateProductWeight($this->id);
         }
 
-        return self::$_totalWeight[(int) $this->id];
+        return (float) (self::$_totalWeight[(int) $this->id] ?? 0);
```

Comment:
```suggestion
        return self::$_totalWeight[(int) $this->id] ?? 0;
```

Float conversation has no aim here :thinking: